utilities/service_enable_condor_poller

```diff
@@ -74,6 +74,10 @@ is_a_newer_than_b () {
 			break
 		fi
 	done
+	# if the first $min entries in A and B are equal and A is longer, A is newer
```

Comment:
What do we want to achieve here?

Reply:
This part handles version numbers that have the same pattern for the first $min components. For example up to this point there is no difference found between 2.0 and 2.0.1 because we only compare as many components as are in the shortest version number. This statement states that in this case, if the first array is longer, it is newer.

Reply:
Right, but does it make sense to always use the newer one?
It seems to make sense in the case of 2.0 vs 2.0.1 (which hopefully never happens since that would be a big change in the way the version is declared).
But does it make sense to use the newer one in 2.0.1 vs 2.0.1dev0 ?   I'm not sure we should install a dev version but only stable releases.
Also, I'm not sure "2.0.1.dev0" would be a newer version than 2.0.1 or just the initial dev version for a future 2.0.1 (in which case it would even be older).

Reply:
This part should only compare numeric version number components. The non-numeric parts are chopped off on line 59 and 62 through parse_version. 

A dev version would only be installed if we specified it around line 225. In this script, when we detect the user's package is newer than the one we specified, they receive a warning. When the package we specify is newer than the user's we attempt to upgrade it using the name and unparsed version number we specify.